dkulib/nlp/text_cleaner.py

```diff
@@ -0,0 +1,231 @@
+            if token_attributes:
+                if self.keep_filtered_tokens:
+                    token_attribute = token_attributes[0]
+                    if len(token_attributes) > 1 and "is_stop" in token_attributes:  # rare case of numbers in stopwords
+                        token_attribute = [t for t in token_attributes if t != "is_stop"][0]
+                    output[token_attribute] += token.lower_ if self.lowercase else token.text
+                    try:
+                        if token.whitespace_ or token.nbor().is_punct or token.nbor().is_space:
+                            output[token_attribute] += " "
+                    except IndexError:  # when reaching the end of the document, nbor() fails
+                        pass
+            else:
+                cleaned_token = self.clean_token(token)
+                if cleaned_token:
+                    output["cleaned"] += cleaned_token
+                try:
+                    if (
+                        token.whitespace_
+                        or ("is_punct" in self.token_filters and token.nbor().is_punct)
+                        or (self.lemmatization and token.nbor().idx == token.idx + len(token))
+                    ):
+                        output["cleaned"] += " "
+                except IndexError:  # when reaching the end of the document, nbor() fails
+                    pass
+        for k in output:
+            output[k] = re.sub(WHITESPACE_REGEX, " ", output[k]).strip()  # remove multiple spaces
+        return output
+
+    def clean_df(
+        self, df: pd.DataFrame, text_column: AnyStr, language_column: AnyStr = "", language: AnyStr = "language_column",
+    ) -> pd.DataFrame:
+        """Public method to clean a text column in a pandas DataFrame, given language information
+
+        Prepare the dataframe with `self._prepare_df_for_cleaning` to obtain a new column with spaCy documents
+        Run `self.clean_document` on all documents with multithreading
+        Format the output dataframe
+
+        Args:
+            df: Input pandas DataFrame
+            text_column: Name of the column containing text data
+            language_column: Name of the column with language codes in ISO 639-1 format
+            language: Language code in ISO 639-1 format
+                If equal to "language_column" this parameter is ignored in favor of language_column
+
+        Returns:
+            Input dataframe with new columns at the end:
+                - Cleaned text after filter, lemmatization, lowercase and unicode normalization steps
+                - One column for each selected `self.token_filters` with a concatenation of filtered tokens
+
+        """
+        self._prepare_df_for_cleaning(df, text_column, language_column, language)
+        start = perf_counter()
+        logging.info(f"Cleaning {len(df.index)} document(s)...")
+        output = [{}] * len(df.index)
```

Comment:
Is this variable used or is it always overridden by line 221?

Reply:
It is always overriden, so we can remove it,

Reply:
d98bf02